### PR TITLE
Job Secrets System Integration 

### PR DESCRIPTION
**Pull request by kensipe**
Job Secrets System Integration 

Summary:
Adding SI test for job secrets for env vars.

JIRA issues:  DCOS_OSS-2338

### CODE REVIEW

File: tests/system/test_root_metronome.py [hunk 1]
```diff
@@ -239,6 +240,26 @@ def test_docker_job():
...
+    secret_name, secret_value = secret_fixture
+
+    client = metronome.create_client()
+    job_id = uuid.uuid4().hex
+    job_def = common.job_with_secrets(job_id, secret_name=secret_name)
+    with job(job_def):
+        client.run_job(job_id)
+
+        @retry(wait_fixed=1000, stop_max_delay=5000)
+        def task_exist():
```
Comment (alenkacz) [inline tests/system/test_root_metronome.py]: this is just a method definition, right? Don't you have to call this method?

File: tests/system/test_root_metronome.py [hunk 2]
```diff
@@ -239,6 +240,26 @@ def test_docker_job():
...
+            assert len(client.get_runs(job_id)) == 1
+            run_id = client.get_runs(job_id)[0]['id']
+            stdout, stderr, return_code = shakedown.run_dcos_command("task log {} secret-env".format(run_id))
+            assert secret_value == stdout.rstrip()
```
Comment (alenkacz) [inline tests/system/test_root_metronome.py]: please add some meaningful string description for the asserts to ease the debugging when assert fails

File: tests/system/test_root_metronome.py [hunk 3]
```diff
@@ -239,6 +240,26 @@ def test_docker_job():
...
+
+    client = metronome.create_client()
+    job_id = uuid.uuid4().hex
+    job_def = common.job_with_secrets(job_id, secret_name=secret_name)
+    with job(job_def):
+        client.run_job(job_id)
+
+        @retry(wait_fixed=1000, stop_max_delay=5000)
+        def task_exist():
+            assert len(client.get_runs(job_id)) == 1
```
Comment (alenkacz) [inline tests/system/test_root_metronome.py]: please add some meaningful string description for the asserts to ease the debugging when assert fails

File: tests/system/test_root_metronome.py (hunk 2 above)
Comment (kensipe) [reply tests/system/test_root_metronome.py]: the fact that the secret value doesn't match the secret in the text file isn't meaningful enough?

File: tests/system/test_root_metronome.py (hunk 1 above)
Comment (kensipe) [reply tests/system/test_root_metronome.py]: yeah... I'm not sure what happened... on it... THX!!!

File: tests/system/test_root_metronome.py (hunk 2 above)
Comment (alenkacz) [reply tests/system/test_root_metronome.py]: for me the debugging would be easier if something like this was there: "Looking for secret value in the task logs. Expecting 'x' but found 'y'"

File: tests/system/test_root_metronome.py [hunk 4]
```diff
@@ -239,6 +240,27 @@ def test_docker_job():
...
+    job_def = common.job_with_secrets(job_id, secret_name=secret_name)
+    with job(job_def):
+        client.run_job(job_id)
+
+        @retry(wait_fixed=1000, stop_max_delay=5000)
+        def job_run_has_secret():
+            assert len(client.get_runs(job_id)) == 1, "trigger job should be running"
+            run_id = client.get_runs(job_id)[0]['id']
+            stdout, stderr, return_code = shakedown.run_dcos_command("task log {} secret-env".format(run_id))
+            assert secret_value == stdout.rstrip(), "secret value in stdout log Incorrect or missing"
```
Comment (alenkacz) [inline tests/system/test_root_metronome.py]: please also add the value that we found there (so stdout.rstrip())

and Incorrect should be lowercase

**Comment by kensipe** [comment]
failure is unrelated

**Pull request merged**